### PR TITLE
Merging bitmap keys deduplicated

### PR DESCRIPTION
**Pull request by xtonik**
### SUMMARY
Some pieces of code deduplication by introducing routine for bitmap keys intersection. Also some other code was deduplicated using existing routine `BitSetUtil.arrayContainerBufferOf()` . 

### Automated Checks

- [x] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [x] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

File: RoaringBitmap/src/main/java/org/roaringbitmap/Util.java [hunk 1]
```diff
@@ -1201,6 +1201,31 @@ public static void partialRadixSort(int[] data) {
       System.arraycopy(copy, 0, data, 0, data.length);
     }
   }
+
+  /**
+   * It makes intersection of the containers' keys between given bitmaps.
```
Comment (lemire) [inline RoaringBitmap/src/main/java/org/roaringbitmap/Util.java:1206]:
```suggestion
   * It computes the intersection of the containers' keys between given bitmaps.
```

File: RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferUtil.java [hunk 2]
```diff
@@ -1063,6 +1064,40 @@ protected static int unsignedUnion2by2(
     // return pos;
   }
 
+  /**
+   * It makes intersection of the containers' keys between given bitmaps.
```
Comment (lemire) [inline RoaringBitmap/src/main/java/org/roaringbitmap/buffer/BufferUtil.java:1068]:
```suggestion
   * It computes the intersection of the containers' keys between given bitmaps.
```

**Comment by lemire** [comment]
Running tests.

**Comment by xtonik** [comment]
Please rerun the tests, their failure was caused by network error during downloading Gradle.

**Comment by lemire** [comment]
Re-running tests.

**Comment by lemire** [comment]
Will be part of the next release.

**Pull request merged**